File: distinct-powers/problem.js
function howManyDistinctPowers(a_limit, b_limit) {
    return getPowers(a_limit, b_limit).length;
}

// Returns unqiue answers for a ** b
function getPowers(a_limit, b_limit) {
    var powers = new Array();

    for(var a = 2; a <= a_limit; a++) {
        for(var b = 2; b <= b_limit; b++) {
            var power = a ** b;
            if(!(powers.includes(power))) {
                powers.push(power);
            }
        }
    }
    powers = powers.sort(function(a,b) {return a-b});
    return powers;
}

console.log(howManyDistinctPowers(100, 100)); //Answer: 9220

File: distinct-powers/problem.rb
def how_many_distinct_powers(a_limit, b_limit)
    a, b = 2, 2
    powers = []
    while a <= a_limit
        while b <= b_limit
            power = a ** b
            if not powers.include?(power)
                powers << power
            end
            b += 1
        end
        b = 2
        a += 1
    end
    powers.sort.length
end


p how_many_distinct_powers(100, 100)
